### DISCUSSION TITLE
Admin UI not working - "Invalid namespace"

### DISCUSSION THREAD

**Discussion by rChaoz** [socketio/socket.io#4981] [Q&A] [answered]
I'm successfully using Socket.IO in my project. Next, I tried using the Admin UI. I followed all of the steps:
- enable CORS in the server
- call `instrument(...)`
- download [the UI](https://github.com/socketio/socket.io-admin-ui/tree/main/ui/dist) and serve it using `npx serve`

When trying to connect to my Socket.IO server, I get this:

![image](https://github.com/socketio/socket.io/assets/8215371/e13c84ad-90dc-4f89-a9d2-6142f81cc250)

![image](https://github.com/socketio/socket.io/assets/8215371/3fa7ceb4-da09-425f-ab77-3175f5b5074e)

This is shown regardless of the credentials being correct or not.

I'm using the default path and namespace. As indicated, I tried the following troubleshooting steps:
- using Chrome instead
- run `curl "<the server URL>/socket.io/?EIO=4&transport=polling"`, I get the correct response
- I use the latest version for both, so versions shouldn't be an issue
- the server sends the correct CORS headers
- I tried looking at the error object in the `on("connect(ion)_error")` on the client/server
  - **client** - the message is "invalid namespace" and description/context are not set
  - **server** - the event is never called

This is my network tab, I can't see anything wrong:

![image](https://github.com/socketio/socket.io/assets/8215371/62ae4e6e-96a2-4f03-b1e2-f0d45e872b9b)

Any ideas on where I should look next?

**Comment by darrachequesne** [comment]
Hi! "Invalid namespace" means the namespace was not found on the server side.

You should see a CONNECT packet (`40/admin,`, for the `admin` namespace) in the WebSocket frames, could you please check?

**Reply by rChaoz** [reply]
Hmm, interesting. The issue fixed itself.

**Reply by rChaoz** [reply]
Well, it's happening again. Here are the network events, in order:

1. GET `http://localhost:5173/socket.io/?EIO=4&transport=polling&t=...`
   Payload: none
   Response: 200 `0{"sid":"C0tpAuKojGaKfJybAAAB","upgrades":["websocket"],"pingInterval":25000,"pingTimeout":20000,"maxPayload":1000000}`

2. GET `http://localhost:5173/socket.io/?EIO=4&transport=polling&t=...&sid=C0tpAuKojGaKfJybAAAB`
   Payload: `40/admin,{"username":"admin","password":"admin"}`
   Response: 200 `ok`

3. GET `ws://localhost:5173/socket.io/?EIO=4&transport=websocket&sid=C0tpAuKojGaKfJybAAAB`
   Payload: none
   Response: 400 (none)

The only thing that look wrong to me is that the `t` query parameter is missing in the last request. It isn't missing when I watch the same event for my app.

This is what I get when I run it in WebSocket only mode:

![image](https://github.com/socketio/socket.io/assets/8215371/19b89260-8824-4ec8-b663-5e9dcdff4044)

**Reply by rChaoz** [reply]
Also, I had to make [some changes](https://github.com/socketio/socket.io-admin-ui/pull/85) (maybe you can take a look) to the admin UI to be able to use the devalue parser, which I now use. I know that's the the cause of the issue however because switching to the default one doesn't fix it (I was using it for the message above) and the devalue parser works with my app correctly.

**Reply by darrachequesne** [reply]
@rChaoz I was not able to reproduce the issue. Please check the example here: https://github.com/socketio/socket.io-admin-ui/tree/develop/examples/single-server

**Reply by rChaoz** [reply, marked as answer]
Thanks for the help! I am using SvelteKit and I was doing the call to `instrument()` in my `hooks.server.ts` file, but it seems like it's not evaluated until the server receives it's first request. Since the request to /socket.io is caught before it reaches SvelteKit, the hooks file was never evaluated.

**Reply by darrachequesne** [reply]
@rChaoz pheew, great news! Thanks for the update :+1:
